src/wayland/xdg_activation/mod.rs

```diff
@@ -185,9 +198,10 @@ impl XdgActivationState {
     /// instead use the return arguments to handle any initialization of the data you might need and to copy the token.
     pub fn create_external_token(
         &mut self,
-        app_id: impl Into<Option<String>>,
+        data: impl Into<Option<XdgActivationTokenData>>,
     ) -> (&XdgActivationToken, &XdgActivationTokenData) {
```

Comment:
Let's keep the most common path simple as is, and introduce `create_external_token_from_data` or something along those lines instead. This will also avoid braking the API needlessly.